lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -743,8 +743,10 @@ sub _create_download_lists {
 }
 
 sub _schedule_from_yaml ($self, $args, $skip_chained_deps, $include_children, @load_yaml_args) {
-    my $data = eval { load_yaml(@load_yaml_args) };
-    if (my $error = $@) { return {error_message => "Unable to load YAML: $error"} }
+    my $data = do {
+        try { load_yaml(@load_yaml_args) }
+        catch ($e) { return {error_message => "Unable to load YAML: $e"} }
+    };
```

Comment:
```suggestion
    my $data;
    try { $data = load_yaml(@load_yaml_args) }
    catch ($e) { return {error_message => "Unable to load YAML: $e"} }
```

Reply:
I am currious if this suggestion (as well as another one below) is because the catch. there was another incident like this where you suggest the do{} block but here it is the opposite. I assume that it is because of the last statement in catch but I am not sure and the documentation is not exactly clear to me

Reply:
yes, same as my suggestions that go in the same way. The `do` block is only useful if you want to assign a value from *both* the try as well as catch. Here catch calls return so does not assign a value to data